### ISSUE TITLE
expo-av, react-native 0.65.1: unloadAsync - Possible Unhandled Promise Rejection

### ISSUE THREAD

**Issue by victoriaSh** [expo/expo#14405]
### Summary

I've updated react-native and expo-av to the latest version and now for every unloadAsync for the sounds I get warning:
```
Possible Unhandled Promise Rejection (id: 250):
TypeError: undefined is not a function
TypeError: undefined is not a function
    at removeSubscription (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.cosmopanda.videoalphabet&modulesOnly=false&runModule=true:143902:46)
    at remove (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.cosmopanda.videoalphabet&modulesOnly=false&runModule=true:143875:35)
    at anonymous (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.cosmopanda.videoalphabet&modulesOnly=false&runModule=true:148739:26)
    at forEach (native)
    at _clearSubscriptions (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.cosmopanda.videoalphabet&modulesOnly=false&runModule=true:148738:36)
    at unloadAsync$ (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.cosmopanda.videoalphabet&modulesOnly=false&runModule=true:148850:41)
    at call (native)
    at tryCatch (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.cosmopanda.videoalphabet&modulesOnly=false&runModule=true:1429:23)
    at invoke (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.cosmopanda.videoalphabet&modulesOnly=false&runModule=true:1602:32)
...
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android, iOS

### Environment

Expo CLI 4.11.0 environment info:
    System:
      OS: macOS 11.4
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.17.0 - /usr/local/bin/node
      npm: 7.21.0 - /opt/homebrew/bin/npm
      Watchman: 2021.08.23.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
      Android SDK:
        API Levels: 29, 30
        Build Tools: 29.0.2, 30.0.2, 30.0.3
        System Images: android-30 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7678000
      Xcode: 12.5/12E262 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^42.0.3 => 42.0.3 
      react: ^17.0.2 => 17.0.2 
      react-dom: 16.13.1 => 16.13.1 
      react-native: ^0.65.1 => 0.65.1 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.11.0
    Expo Workflow: bare

### Reproducible demo or steps to reproduce from a blank project

My code example: 
```
export default function SplashScreen({ route, navigation}) {

    const [music, setMusic] = useState();

    useEffect(() => {
        _initSound();
        return () => {
            _clearSound(music)
        }
    },[]);

    _initSound = async() => {
        await Audio.setAudioModeAsync({
            playsInSilentModeIOS: true,
        });
        const {sound: music} = await Audio.Sound.createAsync(require('./../sounds/funny.mp3'),
                { shouldPlay: true });
        setMusic(music)
    }

    const _clearSound = async(sound) => {
	if (sound){
		sound.unloadAsync();
	}
    }



}
```

**Comment by SergiOnGit** [comment]
The same here

**Comment by SergiOnGit** [comment]
The same here

**Comment by bbarthec** [comment]
Hello @fivecar 👋 
Thank you for reporting this issue. It's seems to be similar to #14449. Apparently there's some change in `RN@0.65` affecting EventEmitter. [`react-native-unimodules@0.14.9`](https://www.npmjs.com/package/react-native-unimodules/v/0.14.9), [`@unimodules/react-native-adapter@6.3.8`](https://www.npmjs.com/package/@unimodules/react-native-adapter/v/6.3.8) and [`expo@42.0.5`](https://www.npmjs.com/package/expo/v/42.0.5) now contain the respective fix for this regression 🔧 
I'm tentatively closing this issue, but if the problem persists for some reason please be welcome to reopen 😊

**Comment by victoriaSh** [comment]
Unfortunately, that doesn't help. I still get "Unhandled Promise Rejection" for both pauseAsync and unloadAsync and audio keeps playing. Looks like "music" constant became "undefinied" right before "_clearSound" is called. I tried to log "music" object after it began to play and object is not undefined.
I'm stuck on this and will be grateful for any help or temporary solution.

There is my full package.json
```

{
  "main": "index.js",
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "web": "expo start --web",
    "start": "react-native start",
    "test": "jest"
  },
  "dependencies": {
    "@babel/runtime": "^7.12.5",
    "@benjeau/react-native-draw": "^0.3.3",
    "@react-native-async-storage/async-storage": "^1.13.4",
    "@react-native-community/cli-platform-android": "^5.0.1",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/slider": "^3.0.3",
    "@react-native-firebase/app": "^12.7.5",
    "@react-native-firebase/messaging": "^12.7.5",
    "@react-navigation/bottom-tabs": "^5.11.13",
    "@react-navigation/native": "^5.9.8",
    "@react-navigation/stack": "^5.14.7",
    "@reduxjs/toolkit": "^1.5.1",
    "@sentry/react-native": "^3.0.3",
    "@unimodules/react-native-adapter": "^6.3.8",
    "amplitude-js": "^7.4.1",
    "expo": "^42.0.5",
    "expo-av": "^9.2.3",
    "expo-status-bar": "~1.0.2",
    "expo-updates": "~0.3.2",
    "i18n-js": "^3.8.0",
    "kld-intersections": "^0.7.0",
    "react": "^17.0.2",
    "react-dom": "16.13.1",
    "react-native": "^0.65.1",
    "react-native-branch": "^5.0.3",
    "react-native-device-info": "^8.3.1",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-haptic-feedback": "^1.11.0",
    "react-native-iap": "^7.3.0",
    "react-native-linear-gradient": "^2.5.6",
    "react-native-pager-view": "^5.4.1",
    "react-native-rate": "^1.2.4",
    "react-native-reanimated": "^2.3.0-alpha.2",
    "react-native-safe-area-context": "^3.3.2",
    "react-native-screens": "^3.7.0",
    "react-native-svg": "^12.1.1",
    "react-native-unimodules": "^0.14.9",
    "react-native-video": "^5.1.1",
    "react-native-web": "~0.13.12",
    "react-redux": "^7.2.4",
    "redux": "^4.1.0",
    "redux-thunk": "^2.3.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@sentry/wizard": "^1.1.4",
    "babel-jest": "~26.6.3",
    "eslint": "7.14.0",
    "jest": "^26.6.3",
    "metro-react-native-babel-preset": "^0.66.0",
    "react-devtools-core": "^4.13.5",
    "react-native-codegen": "^0.0.7",
    "react-native-rename": "^2.5.0",
    "react-test-renderer": "17.0.2"
  },
  "jest": {
    "preset": "react-native"
  },
  "rnpm": {
    "assets": [
      "./assets/fonts/"
    ]
  },
  "private": true
}


```

I've updated unimodules and expo, deleted node_modules and installed them again, cleared Pods and installed them again too.
I get errors:
```
AVAudioSession_iOS.mm:1243  Deactivating an audio session that has running I/O. All I/O should be stopped or paused prior to deactivating the audio session.
```
and
```
Possible Unhandled Promise Rejection (id: 0):
TypeError: undefined is not an object (evaluating 'music.pauseAsync')
_callee2$@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:111262:56
tryCatch@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:1429:23
invoke@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:1602:32
tryCatch@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:1429:23
invoke@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:1502:30
@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:1532:19
tryCallTwo@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:28287:9
doResolve@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:28451:25
Promise@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:28310:14
callInvokeWithMethodAndArg@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:1531:33
enqueue@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:1536:157
@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:1553:69
_callee2@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:111253:42
@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:217141:18
forEach@[native code]
@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:217140:53
@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:218710:19
invokePassiveEffectCreate@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:18123:32
invokeGuardedCallbackProd@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:4674:21
invokeGuardedCallback@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:4778:42
flushPassiveEffectsImpl@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:18193:36
unstable_runWithPriority@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:54645:30
@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:17978:36
workLoop@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:54596:48
flushWork@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:54571:28
_flushCallback@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:54281:24
@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:28870:26
_callTimer@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:28790:17
callTimers@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:28991:19
__callFunction@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:4113:36
@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:3837:31
__guard@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:4064:15
callFunctionReturnFlushedQueue@http://192.168.1.7:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.cosmopanda.videoalphabet:3836:21
callFunctionReturnFlushedQueue@[native code]
```

**Comment by victoriaSh** [comment]
No need to reopen it. After updating unimodules and expo everything works fine. I have found that was my fault :flushed:
I've used react-native-sound before and tried to clear audio object before component will unmount as I did before. That's completely wrong.
Just reminder to myself and anyone who will face the same error from my last comment - read the documentation carefully.
```
useEffect(()=>{
        return music ? () => {
          music.unloadAsync(); }
        : undefined;
    },[music])

```
